services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/docker_api/_core.py

```diff
@@ -176,9 +181,24 @@ async def _get_service_latest_task(service_id: str) -> Mapping[str, Any]:
         raise
 
 
+_TASK_STATE_TO_PROGRESS_MAPPING: Final[dict[str, float]] = (
+    {
+        "pending": 0.2,
+        "assigned": 0.4,
+        "preparing": 0.6,
+        "starting": 0.8,
+        "running": 1.0,
+    }
+    | dict.fromkeys(TASK_STATES_COMPLETE, 1.0)
+    | dict.fromkeys(TASK_STATES_FAILED, 1.0)
+)
+
+
 async def get_dynamic_sidecar_placement(
     service_id: str,
     dynamic_services_scheduler_settings: DynamicServicesSchedulerSettings,
+    *,
+    progress_update: Callable[[float], Awaitable[None]] | None = None,
```

Comment:
MINOR: If used more than one, I prefer `Protocol` to callables. Then you can even define positional/key arguments